contracts/governed-owner/SimpleGovernance.sol

```diff
@@ -0,0 +1,335 @@
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+// mainly used for governed-owner to do infrequent sgn/cbridge owner operations,
+// relatively prefer easy-to-use over gas-efficiency
+contract SimpleGovernance {
+    uint256 public constant THRESHOLD_DECIMAL = 100;
+    uint256 public constant MIN_ACTIVE_PERIOD = 3600; // one hour
+    uint256 public constant MAX_ACTIVE_PERIOD = 2419200; // four weeks
+
+    using SafeERC20 for IERC20;
+
+    enum ParamName {
+        ActivePeriod,
+        QuorumThreshold, // default threshold for votes to pass
+        FastPassThreshold // lower threshold for less critical operations
+    }
+
+    enum ProposalType {
+        ExternalDefault,
+        ExternalFastPass,
+        InternalParamChange,
+        InternalVoterUpdate,
+        InternalProxyUpdate,
+        InternalTransferToken
+    }
+
+    mapping(ParamName => uint256) public params;
+
+    struct Proposal {
+        bytes32 dataHash; // hash(proposalType, targetAddress, calldata)
+        uint256 deadline;
+        mapping(address => bool) votes;
+    }
+
+    mapping(uint256 => Proposal) public proposals;
+    uint256 public nextProposalId;
+
+    address[] public voters;
+    mapping(address => uint256) public voterPowers; // voter addr -> voting power
+
+    // NOTE: proxies must be audited open-source non-upgradable contracts with following requirements:
+    // 1. Truthfully pass along tx sender who called the proxy function as the governance proposer.
+    // 2. Do not allow arbitary fastpass proposal with calldata constructed by the proxy callers.
+    // See ./proxes/CommonOwnerProxy.sol for example.
+    mapping(address => bool) public proposerProxies;
+
+    address deployer;
+
+    event Initiated(
+        address[] voters,
+        uint256[] powers,
+        address[] proxies,
+        uint256 activePeriod,
+        uint256 quorumThreshold,
+        uint256 fastPassThreshold
+    );
+
+    event ProposalCreated(
+        uint256 proposalId,
+        ProposalType proposalType,
+        address target,
+        bytes data,
+        uint256 deadline,
+        address proposer
+    );
+    event ProposalVoted(uint256 proposalId, address voter, bool vote);
+    event ProposalExecuted(uint256 proposalId);
+    event ProposalExecutionReverted(string reason);
+
+    event ParamChangeProposalCreated(uint256 proposalId, ParamName name, uint256 value);
+    event VoterUpdateProposalCreated(uint256 proposalId, address[] voters, uint256[] powers);
+    event ProxyUpdateProposalCreated(uint256 proposalId, address[] addrs, bool[] ops);
+    event TransferTokenProposalCreated(uint256 proposalId, address receiver, address token, uint256 amount);
+
+    constructor() {
+        deployer = msg.sender;
+    }
+
+    /*********************************
+     * External and Public Functions *
+     *********************************/
+
+    // initialize configs
+    function init(
```

Comment:
Is this meant to be upgradable?

Reply:
intend was to make inheritance simpler when the design pattern was different. Will move this to constructor